Kitodo-DataManagement/src/main/resources/db/migration/V2_40__Change_projectfilegroup_to_folder.sql

```diff
@@ -0,0 +1,92 @@
+--
+-- (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+--
+-- This file is part of the Kitodo project.
+--
+-- It is licensed under GNU General Public License version 3 or later.
+--
+-- For the full copyright and license information, please read the
+-- GPL3-License.txt file that was distributed with this source code.
+--
+
+-- Create columns (boolean) copyFolder, (boolean) createFolder,
+--     (double) derivative, (int) dpi, (double) imageScale, (int) imageSize,
+--     (enum {ALL, EXISTING, NO, PREVIEW_IMAGE}) linkingMode
+
+ALTER TABLE projectFileGroup
+  ADD copyFolder   tinyint(1) NOT NULL DEFAULT 1
+        COMMENT 'whether the folder is copied during export',
+  ADD createFolder tinyint(1) NOT NULL DEFAULT 1
+        COMMENT 'whether the folder is created with a new process',
+  ADD derivative   double              DEFAULT NULL
+        COMMENT 'the percentage of scaling for createDerivative()',
+  ADD dpi          int(11)             DEFAULT NULL
+        COMMENT 'the new DPI for changeDpi()',
+  ADD imageScale   double              DEFAULT NULL
+        COMMENT 'the percentage of scaling for getScaledWebImage()',
+  ADD imageSize    int(11)             DEFAULT NULL
+        COMMENT 'the new width in pixels for getSizedWebImage()',
+  ADD linkingMode  varchar(13) NOT NULL COLLATE utf8mb4_unicode_ci DEFAULT 'ALL'
+        COMMENT 'how to link the contents in a METS fileGrp',
+  ADD CONSTRAINT CK_folder_linkingMode
+        CHECK (linkingMode IN ('ALL', 'EXISTING', 'NO', 'PREVIEW_IMAGE'));
+
+-- Set 'linkingMode' column to 'EXISTING' where 'folder' is not empty
+
+UPDATE projectFileGroup SET linkingMode = 'EXISTING'
+  WHERE id > 0 AND folder <> '';
+
+
+-- Set 'linkingMode' to 'PREVIEW_IMAGE' where 'previewImage' = 1
+
+UPDATE projectFileGroup SET linkingMode = 'PREVIEW_IMAGE'
+  WHERE id > 0 AND previewImage = 1;
+
+
+-- Rename columns 'name' => 'fileGroup', 'path' => 'urlStructure',
+--     'folder' => 'path' (no column with same name as table)
+
+ALTER TABLE projectFileGroup
+  CHANGE name   fileGroup    varchar(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
+  CHANGE path   urlStructure varchar(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
+  CHANGE folder path         varchar(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL;
+
+
+-- Delete columns 'suffix' and 'previewImage'. 'previewImage' is now part of
+--     'linkingMode'; 'suffix' depends on 'mimeType' and needs no extra storage.
```

Comment:
I think removing of column `suffix` is wrong as it contains data which is necessary if data is stored inside one directory and differ only through different names which are created with the help of content of `suffix` column.